src/main.rs

```
@@ -22,15 +22,35 @@ macro_rules! printerr(
 
 struct Options {
     path: String,
-    file: Option<String>
+    possible_alternates_path: Option<String>
+}
+
+fn is_hidden(entry: &DirEntry) -> bool {
+    entry.file_name()
+         .to_str()
+         .map(|s| entry.depth() > 0 && s.starts_with("."))
+         .unwrap_or(false)
 }
 
-fn get_possible_files_from_glob() -> Result<Vec<std::path::PathBuf>, glob::PatternError> {
-    glob("**/*").map(|paths| paths.flat_map(identity).filter(|path| path.is_file()).collect())
+fn get_possible_files() -> Vec<PathBuf> {
+    WalkDir::new(".")
+        .follow_links(true)
+        .into_iter()
+        .filter_entry(|e| !is_hidden(e))
```

@wezm This is the only problem I have with this pull request. By filtering hidden files you are introducing a breaking change from the previous version as the Glob version doesn't filter out hidden files.

Also, I am not sure we want to filter out hidden files all the time anyways. I would be fine merging your pull request if you eleminate the filtering of hidden files.

Going forward though, we should have some sort of mechanism of passing filter patterns in via the command line. We should probably tackle that as a separate pull request though, just to scope the work.

---

I understand the desire to not introduce a breaking change but given that tools like `rg` and `fd`, and `ls` all filter dotfiles by default this feels like the correct behaviour to me. Additionally the "faster" example in the README using `find` also filters out a bunch of dotfiles, suggesting that this is the desired behaviour.

Can make the change if you want, but just want to raise that first.

---

@wezm that is a good point. Maybe it is time for another major version bump and introducing a breaking change.

It probably is the common use case that you want to ignore hidden files. But, I don't want it to prevent people from being able to use it with hidden files if they want.

What do you think about ignoring hidden files by default but having a option like `-a` to include hidden files as well, and then some other option to allow filter patterns?

---

Yeah defaulting to hiding them and adding `-a` include hidden files sounds great to me. I'll try to add that over the weekend.

---

Ok, I did some digging into ripgrep and it is built using a crate called ignore which is built on walkdir. Anyways, that ignore crate recognizes and ignores globs specified in both `.ignore` and `.gitignore` files and hidden files by default, but allows you to include hidden files if you want.

So, maybe we don't need command line options to add filters if we just use the `ignore` crate an the `.ignore` file at the root of projects to ignore more files.

I am happy with just merging this in and interating.